big-o-notation/big-o.js

```diff
@@ -0,0 +1,44 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {};                  // 1 * 1 = O(1)
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n)
+    i++                             // 2 * n = O(2n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n)
+    if (!seen[word]) {              // 2 * n = O(2n)
+      seen[word] = true;
+      unique[unique.length] = word; // 3 * n = O(3n)
+    }
+  }
+  return unique;
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n)
+    i++                             // 2 * n = O(2n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n)
+    let isUnique = true;            // 1 * 1 = O(1)
+    for (
+      let c = 0;                    // 1 * 1 = O(1)
+      c < i;                        // 1 * n = O(n)
+      c++                           // 2 * n = O(2n)
+    ) {
+      const comparing = words[c];   // 2 * n = O(2n)
+      if (comparing === word) {     // 1 * n = O(n)
+        isUnique = false;           // 1 * 1 = O(1)
+      }
```

Comment:
Since all of these statements (31-37) run inside of the outer loop, worst-case times complexity is `O(n^2)`

for example

```js
      const comparing = words[c];   // 2 * n^2 = O(2n^2) ~= O(n^2)
```

Typically, a nested loop through the same input is quadratic.